### ISSUE TITLE
pylint type error

### ISSUE THREAD

**Issue by peach-lasagna** [PyCQA/pylint#3929]
### Steps to reproduce
my code:
```python

class User(BaseModel):
    username: str

class UserInDB(User):
    ...

def get_user(db, username: str):
    if username in db:
        user_dict = db[username]
        return UserInDB(**user_dict)

def authenticate_user(fake_db, username: str, password: str):
    user = get_user(fake_db, username)
    if not user:
        return False
    if not verify_password(password, user.hashed_password):
        return False
    return user

async def login_for_access_token(form_data: OAuth2PasswordRequestForm = Depends()):
    user = authenticate_user(
        fake_users_db, form_data.username, form_data.password
    )
    if not user:
        raise HTTPException()
    data={"sub": user.username}# errors on this line:
    # Instance of 'UserInDB' has no 'username' memberpylint(no-member)
    # Instance of 'bool' has no 'username' memberpylint(no-member) 
```

### Current behavior
- Instance of 'UserInDB' has no 'username' memberpylint(no-member)
- Instance of 'bool' has no 'username' memberpylint(no-member) 

### Expected behavior
No errors

### pylint --version output
pylint 2.6.0
astroid 2.4.2
Python 3.9.0 (default, Oct 13 2020, 20:14:06) 
[GCC 8.3.0]

**Comment by hippo91** [comment]
@peach-lasagna thanks for this report.
I tried to reproduce it and succeeded in. However IMHO this is not a `pylint`'s false positive.
For example, the code below is a simplified version of yours:
```
#pylint:disable=missing-module-docstring, missing-class-docstring
#pylint:disable=missing-function-docstring, invalid-name
#pylint:disable=too-few-public-methods
class User:
    username: str

class UserInDB(User):
    ...

def get_user(db, username):
    if username in db:
        user_dict = db[username]
        return UserInDB()
    return UserInDB()

def authenticate_user(fake_db, username):
    user = get_user(fake_db, username)
    if not user:
        return False
    return user

def login_for_access_token(form_data):
    fake_users_db = {'toto': 'toto roberrto'}
    user = authenticate_user(
        fake_users_db, form_data.username
    )
    data={"sub": user.username}# errors on this line:
    print(data)

class FakeForm:
    def __init__(self, name):
        self.username = name
fake_form = FakeForm("toto")
login_for_access_token(fake_form)
```
Linting it gives:
```
************* Module bug_pylint_3929
bug_pylint_3929.py:12:8: W0612: Unused variable 'user_dict' (unused-variable)
bug_pylint_3929.py:27:17: E1101: Instance of 'UserInDB' has no 'username' member (no-member)
bug_pylint_3929.py:27:17: E1101: Instance of 'bool' has no 'username' member (no-member)
```
Which is coherent with the python execution:
```
python bug_pylint_3929.py 
Traceback (most recent call last):
  File "/home/anaxi/PROGRAMMATION/Python/pylint_tmp/bug_pylint_3929.py", line 34, in <module>
    login_for_access_token(fake_form)
  File "/home/anaxi/PROGRAMMATION/Python/pylint_tmp/bug_pylint_3929.py", line 27, in login_for_access_token
    data={"sub": user.username}# errors on this line:
AttributeError: 'UserInDB' object has no attribute 'username'
```
Note that if i change a bit the `User` class to:
```
class User:
    username: str = None
```
then the execution is ok AND `pylint` does not complain anymore.

However this analysis is limited by the fact that i don't know anything about the `BaseModel` class.

As it seems that `pylint`'s behavior is ok, i close this issue. If i'm wrong i will of course reopen it.
